### DISCUSSION TITLE
sBTC MVP Requirements

### DISCUSSION THREAD

**Discussion by AshtonStephens** [stacks-sbtc/sbtc#3] [Research]
## Summary

***This is a discussion to define the minimum viable product for sBTC, and will break down the features requested into the "must have", "should have", and "nice to have" categories.*** Due to time constraints and the high stakes for correctness this determination will be brutal, so some features that to some would put in "should" or "must" have may fall into the "nice to have" category for sBTC-v1.

This post only begins the discussion and is not a complete assessment, and is subject to change as the discussion unfolds.

## Features

![iteration count is 5](https://img.shields.io/badge/iterations-5-blue)

### Must Have
1. sBTC is a SIP-10 token backed by 1-1 by a Bitcoin peg wallet
1. The sBTC peg wallet is maintained and managed by the set of sBTC signers
1. The signer binary can screen for a dynamic blocklist effectively
1. An initial community vote by STX weight determines the sBTC signing set
1. Bitcoin can be converted into sBTC within 6 Bitcoin blocks
1. sBTC can be converted into Bitcoin within 3 Bitcoin blocks
1. SIP-10 token contact stays the same across sBTC releases
1. STX for gas fees are easily available - a faucet can be an MVP option
1. The sBTC signing set persists for the number of months expected to release sBTC v2
1. Signers can rotate their private keys

### Should have
1. Activating sBTC does not require a hard fork
1. All client facing contracts stay the same across mainnet activations of sBTC versions
1. STX gas fees can be effectively paid with sBTC via a wrapper https://github.com/stacks-network/stacks-core/issues/4235
1. Bitcoin can be converted into sBTC within 1 Bitcoin block
1. sBTC can be converted into Bitcoin within 1 Bitcoin block
1. Deposits and Withdrawals can be initiated from either the burnchain or the stacks blockchain (must be in v2)

**Comment by AshtonStephens** [comment]
Note: make a discussion for https://github.com/stacks-network/stacks-core/issues/4235

**Comment by AshtonStephens** [comment]
Ran this by the team and then Andre - looks good.

**Comment by AshtonStephens** [comment]
Changed the requirements to say withdrawal can take 6 blocks based on the discussions in #13 where withdrawal will need to wait for the withdrawal request to be sufficiently final before making the withdrawal.

**Comment by AshtonStephens** [comment]
Updated the requirements:
- Remove the release date from must haves
- Include signing key rotation as a must-have per conversations with @andrerserrano .

**Reply by andrerserrano** [reply]
Yes, feedback from signers is that rotation key is "very important feature"
